### ISSUE TITLE
Self-referencing comparison misbehaves with matching object names

### ISSUE THREAD

**Issue by yegor256** [objectionary/lints#654]
This is my code:

```
# Example.
[] > app
  a. > foo
    b > foo1!
```

I'm getting this:

```
The object "$.foo1.a" can't copy itself (self-referencing (Single))
```

Then, I do this:

```
# Example.
[] > app
  a. > foo
    b > bar!
```

And I get no warnings. It seems, we compare strings incorrectly.

**Comment by VasilevNStas** [comment]
@h1alexbel @yegor256 I checked both cases from the report against the current `critical/self-referencing` lint and neither produces a defect anymore.

The root cause of the original false positive was the parser auto-naming: with `b > foo1!`, the parser used to name the object `foo1` (from `foo1!`), so the copied object became `<o name="foo1" base="ξ.foo1.a">` and the lint matched `base = concat(ξ., @name)` (i.e. `ξ.foo1`). Now the same code produces `<o name="foo" base="ξ.foo1.a">` — the object is named `foo`, `base` is `ξ.foo1.a`, which does not equal `ξ.foo`, so no defect. The `bar!` case behaves identically.

Since the parser changed its auto-naming behavior, the original misbehavior is gone on `master`. I'd suggest closing this as not reproducible; if the comparison itself ever needs tightening, the lint could switch from exact `concat` to a proper ancestor-aware check, but there is nothing to fix right now.
